File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> hashMap; 
        for(int n : nums){
            hashMap[n]++; 
        }
        vector<int> answer; 
        priority_queue<pair<int,int>, vector<pair<int,int>>> pq; 
        for(auto& it : hashMap){
            pq.push({it.second, it.first}); 
        }
        while(k--){
            answer.push_back(pq.top().second);
            pq.pop(); 
        }
        return answer; 
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> hashMap = new HashMap<>(); 
        for(int i = 0; i < nums.length; i++){
            hashMap.put(nums[i], hashMap.getOrDefault(nums[i],0) + 1); 
        }

        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b)->{
            return hashMap.get(b) - hashMap.get(a); 
        });

        for(Integer n : hashMap.keySet()){
            pq.offer(n); 
        }

        List<Integer> answer = new ArrayList<>(); 
        while(k > 0){
            answer.add(pq.poll()); 
            k--; 
        }

        return answer.stream().mapToInt(Integer::intValue).toArray(); 
    }
}